src/wallet/rpcwallet.cpp

```diff
@@ -239,6 +245,56 @@ UniValue getaddressinfo(const UniValue& params, bool fHelp)
     return ret;
 }
 
+void parseWIFKey(std::string strKey, CWallet* pwallet, CPubKey& master_pub_key) {
+    CKey key = KeyIO::DecodeSecret(strKey);
+    if (!key.IsValid()) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid private key");
+    }
+
+    if (HaveKey(pwallet, key)) {
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Already have this key (either as an HD seed or as a loose private key)");
+    }
+    master_pub_key = pwallet->GetScriptPubKeyMan()->DeriveNewSeed(key);
+}
+
+UniValue upgradewallet(const UniValue& params, bool fHelp) {
```

Comment:
Please use a newline for the opening bracket of the function body `{`.
Same in `parseWIFKey`.